Mathlib/Analysis/NormedSpace/AffineIsometry.lean

```diff
@@ -0,0 +1,933 @@
+  { coe := fun f => f.toFun
+    inv := fun f => f.invFun
+    left_inv := fun f => f.left_inv
+    right_inv := fun f => f.right_inv,
+    coe_injective' := fun f g h _ => by
+      cases f
+      cases g
+      congr
+      simpa [FunLike.coe_injective.eq_iff] using h }
+
+@[simp]
+theorem coe_mk (e : P ≃ᵃ[𝕜] P₂) (he : ∀ x, ‖e.linear x‖ = ‖x‖) : ⇑(mk e he) = e :=
+  rfl
+#align affine_isometry_equiv.coe_mk AffineIsometryEquiv.coe_mk
+
+@[simp] -- simp progress
+theorem coe_toAffineEquiv (e : P ≃ᵃⁱ[𝕜] P₂) : ⇑e.toAffineEquiv = e :=
+  rfl
+#align affine_isometry_equiv.coe_to_affine_equiv AffineIsometryEquiv.coe_toAffineEquiv
+
+theorem toAffineEquiv_injective : Injective (toAffineEquiv : (P ≃ᵃⁱ[𝕜] P₂) → P ≃ᵃ[𝕜] P₂)
+  | ⟨_, _⟩, ⟨_, _⟩, rfl => rfl
+#align affine_isometry_equiv.to_affine_equiv_injective AffineIsometryEquiv.toAffineEquiv_injective
+
+@[ext]
+theorem ext {e e' : P ≃ᵃⁱ[𝕜] P₂} (h : ∀ x, e x = e' x) : e = e' :=
+  toAffineEquiv_injective <| AffineEquiv.ext h
+#align affine_isometry_equiv.ext AffineIsometryEquiv.ext
+
+/-- Reinterpret a `AffineIsometryEquiv` as a `AffineIsometry`. -/
+def toAffineIsometry : P →ᵃⁱ[𝕜] P₂ :=
+  ⟨e.1.toAffineMap, e.2⟩
+#align affine_isometry_equiv.to_affine_isometry AffineIsometryEquiv.toAffineIsometry
+
+@[simp] -- simp progress
+theorem coe_toAffineIsometry : ⇑e.toAffineIsometry = e :=
+  rfl
+#align affine_isometry_equiv.coe_to_affine_isometry AffineIsometryEquiv.coe_toAffineIsometry
+
+/-- Construct an affine isometry equivalence by verifying the relation between the map and its
+linear part at one base point. Namely, this function takes a map `e : P₁ → P₂`, a linear isometry
+equivalence `e' : V₁ ≃ᵢₗ[k] V₂`, and a point `p` such that for any other point `p'` we have
+`e p' = e' (p' -ᵥ p) +ᵥ e p`. -/
+def mk' (e : P₁ → P₂) (e' : V₁ ≃ₗᵢ[𝕜] V₂) (p : P₁) (h : ∀ p' : P₁, e p' = e' (p' -ᵥ p) +ᵥ e p) :
+    P₁ ≃ᵃⁱ[𝕜] P₂ :=
+  { AffineEquiv.mk' e e'.toLinearEquiv p h with norm_map := e'.norm_map }
+#align affine_isometry_equiv.mk' AffineIsometryEquiv.mk'
+
+@[simp] -- simp progress
+theorem coe_mk' (e : P₁ → P₂) (e' : V₁ ≃ₗᵢ[𝕜] V₂) (p h) : ⇑(mk' e e' p h) = e :=
+  rfl
+#align affine_isometry_equiv.coe_mk' AffineIsometryEquiv.coe_mk'
+
+@[simp]
+theorem linearIsometryEquiv_mk' (e : P₁ → P₂) (e' : V₁ ≃ₗᵢ[𝕜] V₂) (p h) :
+    (mk' e e' p h).linearIsometryEquiv = e' := by
+  ext
+  rfl
+#align affine_isometry_equiv.linear_isometry_equiv_mk' AffineIsometryEquiv.linearIsometryEquiv_mk'
+
+end AffineIsometryEquiv
+
+namespace LinearIsometryEquiv
+
+variable (e : V ≃ₗᵢ[𝕜] V₂)
+
+/-- Reinterpret a linear isometry equiv as an affine isometry equiv. -/
+def toAffineIsometryEquiv : V ≃ᵃⁱ[𝕜] V₂ :=
+  { e.toLinearEquiv.toAffineEquiv with norm_map := e.norm_map }
+#align linear_isometry_equiv.to_affine_isometry_equiv LinearIsometryEquiv.toAffineIsometryEquiv
+
+-- @[simp] -- Porting note: simp-normal form is `coe_toAffineIsometryEquiv'`
+theorem coe_toAffineIsometryEquiv : ⇑(e.toAffineIsometryEquiv : V ≃ᵃⁱ[𝕜] V₂) = e :=
```

Comment:
What does this porting note mean?

Reply:
That should be deleted. `coe_toAffineIsometryEquiv’` was provable by `simp` and removed

Reply:
Done.